### DISCUSSION TITLE
Changing USB cable causes different behaviours in Tiny USB

### DISCUSSION THREAD

**Discussion by denybear** [hathach/tinyusb#2136] [Q&A]
Hi,
I am using a midi host implementation from a merge request ([merge request](https://github.com/hathach/tinyusb/pull/1627)).
My target is a Raspberry pico (midi host), connected to Novation Launchpad mini (MKI).

My raspi pico is connected to launchpad using 2 cables: pico--cable A--cable B (or C)--Launchpad
- Cable A: From Raspi pico to micro USB, type B male, OTG to USB type A female
- Cable B is connecting Cable A female USB to Novation Launchpad, so:  USB type A male to micro USB, type B, non-OTG.
- Cable C is supposedly identical to cable B (at least my multimeter says so; same connections to the same pinouts).

My findings:

- With cable A + B, my program with midi host implementation works fine and I can send some midi messages to my Launchpad device (some traces : [success.txt](https://github.com/hathach/tinyusb/files/11931203/success.txt))

- With cable A + C, which provides the same pin connections, USB communication stops abruptly during device enumeration (some traces: [fail.txt](https://github.com/hathach/tinyusb/files/11931202/fail.txt)). From what it seems, issue is timeout in hcd_rp2040.c, line 234 onwards.

Am I doing anything wrong? Have you ever seen similar behaviour, ie. changing USB cable affects behaviour of tiny USB?
Would you have any idea/recommendation to help me solve (or at least understand) this issue? 
Thanks!

**Comment by JustAnother1** [comment]
Could this be a broken cable (unreliable connection)? Lost data would explain why the communication stops.
